Thanks for the contribution.
Linux build fails currently:
```
../src/frmmain_settings.cpp: In member function ‘void frmMain::loadSettings()’:
../src/frmmain_settings.cpp:151:56: error: ambiguous overload for ‘operator=’ (operand types are ‘QStringList’ and ‘<brace-enclosed initializer list>’)
  151 |         jogSteps = {"0.01", "0.1", "1", "5", "10", "50"};
```
Could you talk a look at it?

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -133,8 +133,67 @@ jobs:
           name: ${{ env.ARTIFACT }}
           path: ${{ runner.temp }}\${{ env.ARTIFACT }}
 
+  build_macos:
+    # Builds an arm64-only .app bundle on Apple Silicon. Intel Mac users need
+    # to build from source — Candle2's pre-existing CI didn't ship macOS at all,
+    # so any automated build is progress. A universal-binary step can be added
+    # later without restructuring the job.
+    runs-on: macos-latest
+
+    env:
+      SOURCE_DIR: ${{ github.workspace }}
+      ARTIFACT:   candle2-macos.zip
+
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+
+      - name: Install Qt6
+        # Homebrew's `qt` formula is Qt 6.x. `qtserialport` is a separate Qt6
+        # module (unlike Qt5 where it was part of qt-everywhere). Both are
+        # needed to build Candle2 on macOS.
+        run: |
+          brew install qt qtserialport
+
+      - name: Configure
+        run: |
+          mkdir build && cd build
+          qmake6 ../src/candle2.pro
+
+      - name: Build
+        run: |
+          cd build
+          make -j$(sysctl -n hw.ncpu)
+
+      - name: Bundle Qt frameworks
+        # macdeployqt rewrites the .app's rpaths and copies all required Qt
+        # frameworks + plugins into Candle2.app/Contents/Frameworks so the
+        # bundle is self-contained. Without this, the app only runs on
+        # machines that already have Homebrew Qt installed.
+        run: |
+          cd build
+          "$(brew --prefix qt)/bin/macdeployqt" Candle2.app -verbose=1
+
+      - name: Zip .app
+        # Use `ditto -k` instead of `zip` so macOS extended attributes
+        # (including the ad-hoc code signature macdeployqt applies) survive
+        # the round-trip through the Actions artifact store.
+        run: |
+          cd build
+          ditto -c -k --sequesterRsrc --keepParent Candle2.app "${{ env.ARTIFACT }}"
+
+      - name: Save build artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ env.ARTIFACT }}
+          path: build/${{ env.ARTIFACT }}
+
   release:
-    needs: [build_linux, build_win]
+    needs: [build_linux, build_win, build_macos]
     runs-on: ubuntu-latest
     permissions:
       contents: write
@@ -165,3 +224,4 @@ jobs:
           files: |
             Candle2_AppImage/Candle2*.AppImage
             candle2-win.zip/candle2-win.zip
+            candle2-macos.zip/candle2-macos.zip

--- a/src/drawers/shaderdrawable.cpp
+++ b/src/drawers/shaderdrawable.cpp
@@ -170,7 +170,7 @@ void ShaderDrawable::draw(QOpenGLShaderProgram *shaderProgram)
     if (!m_triangles.isEmpty()) {
         if (m_texture) {
             m_texture->bind();
-            shaderProgram->setUniformValue("texture", 0);
+            shaderProgram->setUniformValue("u_texture", 0);
         }
         glDrawArrays(GL_TRIANGLES, 0, m_triangles.count());
     }

--- a/src/frmmain.ui
+++ b/src/frmmain.ui
@@ -86,6 +86,16 @@ QTableView {
 	selection-color: rgb(0, 0, 0);
 }
 
+/* QLCDNumber (work coordinate displays) — force a high-contrast color pair
+   because the default palette-derived segment color is nearly indistinguishable
+   from the widget background on macOS Fusion.
+*/
+QLCDNumber {
+	background-color: #f4f4f4;
+	color: #111111;
+	border: 1px solid #999999;
+}
+
 QSlider::horizontal {
 	padding-top: 1px;
 	padding-bottom: 1px;

--- a/src/main.cpp
+++ b/src/main.cpp
@@ -38,6 +38,16 @@ int main(int argc, char *argv[])
     QSurfaceFormat fmt;
     fmt.setSamples(8);
     fmt.setDepthBufferSize(24);
+#ifdef Q_OS_MACOS
+    // macOS only exposes OpenGL 2.1 in the default compatibility profile.
+    // The visualizer's shaders use GLSL 1.50 features (in/out keywords, etc.)
+    // which require a 3.2+ Core context — on macOS that must be requested
+    // explicitly, or you end up with a blank visualizer and shader compile
+    // errors. Linux/Windows drivers provide 3.2+ in their compatibility
+    // profile by default, so no change needed there.
+    fmt.setVersion(3, 2);
+    fmt.setProfile(QSurfaceFormat::CoreProfile);
+#endif
     QSurfaceFormat::setDefaultFormat(fmt);
 
 #ifdef UNIX

--- a/src/shaders/fshader.glsl
+++ b/src/shaders/fshader.glsl
@@ -1,11 +1,14 @@
-#version 130
+#version 150 core
 #define DATA_TYPE_LINE 0
 #define DATA_TYPE_DASH 1
 #define DATA_TYPE_DASH_DOT 2
 #define DATA_TYPE_POINT 3
 #define DATA_TYPE_TRIANGLE 4
 
-uniform sampler2D texture;
+// Renamed from `texture` so it doesn't shadow the GLSL 1.50 built-in
+// `texture()` function. Keeping the old name compiles on some drivers
+// (GLSL scoping lets the call resolve by signature) but fails on others.
+uniform sampler2D u_texture;
 
 in vec4 v_color;
 in vec2 v_position;
@@ -37,6 +40,6 @@ void main()
     } else if (type == DATA_TYPE_POINT) {
         fragColor = v_color;
     } else if (type == DATA_TYPE_TRIANGLE) {
-        fragColor = texture2D(texture, v_texture);
+        fragColor = texture(u_texture, v_texture);
     }
 }

--- a/src/shaders/vshader.glsl
+++ b/src/shaders/vshader.glsl
@@ -1,4 +1,4 @@
-#version 130
+#version 150 core
 #define DATA_TYPE_LINE 0
 #define DATA_TYPE_DASH 1
 #define DATA_TYPE_DASH_DOT 2

